### PR TITLE
Feature/37153: Add custom_image to code_editor_app_settings in Sagemaker user profile and domain

### LINKED ISSUE #37153: [Enhancement]: SageMaker Studio now supports custom images for Code Editor Applications
### Description

Similarly to JupyterLab and Kernel Gateway apps, AWS Sagemaker now supports using custom images for Code Editor applications, see the docs [here](https://docs.aws.amazon.com/sagemaker/latest/dg/code-editor-custom-images-provide-access.html).

This would be a relatively quick addition to add in the aws_sagemaker_domain resource.

### Affected Resource(s) and/or Data Source(s)

aws_sagemaker_domain

### Potential Terraform Configuration

```terraform
resource "aws_sagemaker_domain" "code_editor_domain" {

  default_user_settings {

    code_editor_app_settings {
      custom_image = [
        app_image_config_name = aws_sagemaker_app_image_config.example.app_image_config_name
        image_name            = aws_sagemaker_image_version.example.image_name
        image_version         = 1
      ]
    }
  }
}
```

### References

1. https://docs.aws.amazon.com/sagemaker/latest/dg/code-editor-custom-images-provide-access.html
2. https://aws.amazon.com/about-aws/whats-new/2024/04/amazon-sagemaker-studio-code-editor-custom-images/

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by michaeljneely**
### Description
AWS Sagemaker supports custom images for CodeEditor applications.

Custom images can be added via `aws_sagemaker_user_profile.user_settings.code_editor_app_settings` or `aws_sagemaker_domain.default_user_settings.code_editor_app_settings`

### Relations
Closes #37153 

### References
1. https://aws.amazon.com/about-aws/whats-new/2024/04/amazon-sagemaker-studio-code-editor-custom-images/
2. https://docs.aws.amazon.com/sagemaker/latest/dg/code-editor-custom-images-provide-access.html

### Output from Acceptance Testing
I've added tests but can't run them myself. I would be grateful if someone could confirm these tests pass successfully.

### CODE REVIEW

**Review by DrFaust92 — APPROVED** [review]
LGTM

File: internal/service/sagemaker/domain.go [hunk 1]
```diff
@@ -389,6 +389,27 @@ func ResourceDomain() *schema.Resource {
 											ValidateFunc: verify.ValidARN,
 										},
 									},
+									"custom_image": {
```
Comment (deepakbshetty) [inline internal/service/sagemaker/domain.go:392]: Hi @michaeljneely , I am unable to see expandDomainCustomImages under expandDomainCodeEditorAppSettings & flattenDomainCustomImages under flattenDomainCodeEditorAppSettings.

Comment (michaeljneely) [reply internal/service/sagemaker/domain.go:392]: Thanks, for catching this. Those missing elements have been added.

Comment (deepakbshetty) [reply internal/service/sagemaker/domain.go:392]: LGTM

**Review by DrFaust92 — APPROVED** [review]
LGTM

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSageMaker_serial/Domain/codeEditorAppSettings\|TestAccSageMaker_serial/UserProfile/codeEditorAppSettings' PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/sagemaker/... -v -count 1 -parallel 20  -run=TestAccSageMaker_serial/Domain/codeEditorAppSettings\|TestAccSageMaker_serial/UserProfile/codeEditorAppSettings -timeout 360m
=== RUN   TestAccSageMaker_serial
=== PAUSE TestAccSageMaker_serial
=== CONT  TestAccSageMaker_serial
=== RUN   TestAccSageMaker_serial/UserProfile
=== RUN   TestAccSageMaker_serial/UserProfile/codeEditorAppSettings_customImage
    user_profile_test.go:283: Environment variable SAGEMAKER_IMAGE_VERSION_BASE_IMAGE is not set
=== RUN   TestAccSageMaker_serial/Domain
=== RUN   TestAccSageMaker_serial/Domain/codeEditorAppSettings_customImage
    domain_test.go:660: Environment variable SAGEMAKER_IMAGE_VERSION_BASE_IMAGE is not set
=== RUN   TestAccSageMaker_serial/Domain/codeEditorAppSettings
--- PASS: TestAccSageMaker_serial (250.85s)
    --- PASS: TestAccSageMaker_serial/UserProfile (0.00s)
        --- SKIP: TestAccSageMaker_serial/UserProfile/codeEditorAppSettings_customImage (0.00s)
    --- PASS: TestAccSageMaker_serial/Domain (250.84s)
        --- SKIP: TestAccSageMaker_serial/Domain/codeEditorAppSettings_customImage (0.00s)
        --- PASS: TestAccSageMaker_serial/Domain/codeEditorAppSettings (250.84s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker	255.470s
```

**Comment by ewbankkit** [comment]
@michaeljneely Thanks for the contribution :tada: :clap:.

**Pull request merged**